pyfar/classes/audio.py

```diff
@@ -1021,7 +1039,122 @@ def power(data: tuple, domain='freq'):
     return _arithmetic(data, domain, _power)
 
 
-def _arithmetic(data: tuple, domain: str, operation: Callable):
+def matrix_multiplication(
+        data: tuple, domain='freq', axes=[(-2, -1), (-2, -1), (-2, -1)]):
+    """Matrix multiplication of multidimensional pyfar audio objects and/or
+    array likes.
+
+    This function is based on ``numpy.matmul`` and only acts on the channels,
+    not on the frequency/time dimension. Pyfar audio objects are:
+    :py:func:`Signal`, :py:func:`TimeData`, and :py:func:`FrequencyData`.
+
+    Parameters
+    ----------
+    data : tuple of the form (data_1, data_2, ..., data_N)
+        Data to be multiplied. Can contain pyfar audio objects or array likes.
+        The data is `broadcasted
+        <https://numpy.org/doc/stable/user/basics.broadcasting.html>`_
+        to the data with the highest number of channel dimensions. Pyfar audio
```

Comment:
maybe explicitly mention that this means `len(data.cshape)` (or is it `prod(data.cshape)`)?